homeassistant/components/habitica/__init__.py

```diff
@@ -189,11 +277,21 @@ async def handle_api_call(call: ServiceCall) -> None:
             DOMAIN, SERVICE_API_CALL, handle_api_call, schema=SERVICE_API_CALL_SCHEMA
         )
 
+    if not hass.services.has_service(DOMAIN, "cast_skill"):
+        hass.services.async_register(
+            DOMAIN,
+            "cast_skill",
+            cast_skill,
+            schema=SERVICE_CAST_SKILL_SCHEMA,
+            supports_response=SupportsResponse.ONLY,
+        )
```

Comment:
We should setup services in `async_setup`

Reply:
Because otherwise the availability of services is dependent on a working config entry, and that messes with automation checks